### ISSUE TITLE
[BUG] `MaskedDataset.__getitem__` initializes `y_masked` from `x` instead of `y`

### ISSUE THREAD

**Issue by officialasishkumar** [gc-os-ai/pyaptamer#302]
**Describe the bug**

In `pyaptamer/datasets/dataclasses/_masked.py`, line 154 of `__getitem__` initializes `y_masked` by cloning `x` instead of `y`:

```python
x = torch.tensor(self.x[index], dtype=torch.int64)
y = torch.tensor(self.y[index], dtype=torch.int64)

x_masked = x.clone().detach()
y_masked = x.clone().detach()  # BUG: should be y.clone().detach()
```

This means the target tensor `y_masked` is a copy of the input sequence rather than the actual target sequence. The non-masked positions are then zeroed out from the wrong base tensor.

**To Reproduce**

```python
from pyaptamer.datasets.dataclasses import MaskedDataset

# Use different x and y to make the bug visible
x = [[1, 2, 3, 4, 5, 0]]
y = [[10, 20, 30, 40, 50, 0]]  # different from x

dataset = MaskedDataset(x, y, max_len=6, mask_idx=99, masked_rate=0.5)
x_masked, y_masked, x_orig, y_orig = dataset[0]

# y_masked is derived from x, not y — the values 10, 20, 30, 40, 50 never appear in y_masked
```

**Expected behavior**

`y_masked` should be initialized from `y`, not `x`:
```python
y_masked = y.clone().detach()
```

**Additional context**

In typical masked language modeling (MLM) setups, `x` and `y` are often identical (the same sequence), so this bug may not have been caught in practice. However, the API explicitly accepts separate `x` and `y` arrays, and the variable is named `y_masked`, so it should clone `y`.

Additionally, `MaskedDataset` currently has no dedicated unit tests for `__getitem__`, which is why this went undetected.

**Versions**
<details>
0.1.0a1
</details>

**Comment by siddharth7113** [comment]
cc @NennoMP

**Comment by NennoMP** [comment]
@officialasishkumar Actually, the `__getitem__` method logic is the way it is because the original repository had `y_masked` cloned from `x`. See `Masked_Dataset` class in [https://github.com/PNUMLB/AptaTrans/blob/master/utils.py](https://github.com/PNUMLB/AptaTrans/blob/master/utils.py).

When AptaTrans was implemented we simply tried to replicate and refactor the authors' code. Hard to tell whether this is the intended behaviour or they did a mistake in their codebase.

**Comment by siddharth7113** [comment]
@NennoMP , I am inclined to correct this with a comment about original implementation, the current way doesn't seem to make sense to me, since we are defining `y` and not using it?

**Comment by NennoMP** [comment]
We are using it, the metho returns 4 variables, masked and non-masked. Both are used by the dataloaders.

**Comment by NennoMP** [comment]
It has been some time since I refactored this, so I looked into it. I am inclinded to say this is intended behaviour.

The masked pairs are used to have the model/encoder learn to reconstruct the original sequence (`x` and in this case `y_masked`) by looking at the sequence with masked positions (`x_masked`). Thus, it is correct that `y_masked` clones the original sequence `x`. The other pairs, `x` and `y`, are used for another task entirely. Going by memory these probably contain the input sequence and secondary structure to be predicted, respectively.

These four variables are all used when training AptaTrans' encoders, as they contribute to two different losses (see the corresponding lightning training module `AptaTransEncoderLightning`).

**Comment by siddharth7113** [comment]
> It has been some time since I refactored this, so I looked into it. I am inclinded to say this is intended behaviour.
> 
> The masked pairs are used to have the model/encoder learn to reconstruct the original sequence (`x` and in this case `y_masked`) by looking at the sequence with masked positions (`x_masked`). Thus, it is correct that `y_masked` clones the original sequence `x`. The other pairs are, `x` and `y`, are used for another task entirely. Going by memory these probably contain the input sequence and secondary structure to be predicted, respectively.
> 
> These four variables are all used when training AptaTrans' encoders, as they contribute to two different losses (see the corresponding lightning training module `AptaTransEncoderLightning`).

Yeah, I looked up the pre-trained encoders today, and from previous analysis we know the pretraining is working fine, but the API is having issue, unrelated to this, so for now we can leave it as is.
